src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -1025,122 +1026,181 @@ protected void GenerateFindFirstChar()
             MarkLabel(finishedLengthCheck);
 
             // Generate anchor checks.
-            if ((_anchors & (RegexPrefixAnalyzer.Beginning | RegexPrefixAnalyzer.Start | RegexPrefixAnalyzer.EndZ | RegexPrefixAnalyzer.End)) != 0)
+            if ((_leadingAnchor & (RegexPrefixAnalyzer.Beginning | RegexPrefixAnalyzer.Start | RegexPrefixAnalyzer.EndZ | RegexPrefixAnalyzer.End | RegexPrefixAnalyzer.Bol)) != 0)
             {
-                if (!_code.RightToLeft)
+                switch (_leadingAnchor)
                 {
-                    if ((_anchors & RegexPrefixAnalyzer.Beginning) != 0)
-                    {
-                        Label l1 = DefineLabel();
-                        Ldloc(_runtextposLocal);
-                        Ldthisfld(s_runtextbegField);
-                        Ble(l1);
-                        Br(returnFalse);
-                        MarkLabel(l1);
-                    }
-
-                    if ((_anchors & RegexPrefixAnalyzer.Start) != 0)
-                    {
-                        Label l1 = DefineLabel();
-                        Ldloc(_runtextposLocal);
-                        Ldthisfld(s_runtextstartField);
-                        Ble(l1);
-                        BrFar(returnFalse);
-                        MarkLabel(l1);
-                    }
-
-                    if ((_anchors & RegexPrefixAnalyzer.EndZ) != 0)
-                    {
-                        Label l1 = DefineLabel();
-                        Ldloc(_runtextposLocal);
-                        Ldloc(_runtextendLocal);
+                    case RegexPrefixAnalyzer.Beginning:
+                        if (!_code.RightToLeft)
+                        {
+                            Label l1 = DefineLabel();
+                            Ldloc(_runtextposLocal);
+                            Ldthisfld(s_runtextbegField);
+                            Ble(l1);
+                            Br(returnFalse);
+                            MarkLabel(l1);
+                        }
+                        else
+                        {
+                            Label l1 = DefineLabel();
+                            Ldloc(_runtextposLocal);
+                            Ldloc(_runtextbegLocal!);
+                            Ble(l1);
+                            Ldthis();
+                            Ldloc(_runtextbegLocal!);
+                            Stfld(s_runtextposField);
+                            MarkLabel(l1);
+                            Ldc(1);
```

Comment:
Is there a double `Ldc(1); Reg();` in this code path?

Reply:
Good eye, thanks, I'll remove it.  It doesn't show up in unit tests because it just ends up being dead code.